src/mctpd.c

```diff
@@ -895,6 +924,9 @@ static int cb_listen_monitor(sd_event_source *s, int sd, uint32_t revents,
 
 		case MCTP_NL_DEL_LINK:
 		{
+			link_userdata* lmUserData =
```

Comment:
the userdata is provided in `c->link_userdata`, you won't need a separate lookup here.

(see https://github.com/CodeConstruct/mctp/commit/44cc89bf51ac5ae25db5764c0d4472e252065b4f#diff-3816167d574db6f7e0cd38ed18a7a4f1ffabe3a057a2421b2c364e7668c5202dR34-R37)